lib/rest-api.php

```diff
@@ -18,3 +18,29 @@ function gutenberg_register_global_styles_endpoints() {
 	$global_styles_controller->register_routes();
 }
 add_action( 'rest_api_init', 'gutenberg_register_global_styles_endpoints' );
+
+/**
+ * Adds the post classes to the REST API response.
+ *
+ * @param WP_REST_Response $response    Response object.
+ * @param WP_Post          $post    Post object.
+ *
+ * @return WP_REST_Response Response object.
+ */
+function gutenberg_add_post_class_to_api_response( $response, $post ) {
+	$response->data['post_class'] = get_post_class( '', $post->ID );
+
+	return $response;
+}
+
+/**
+ * Adds the post classes to all post types in the REST API.
+ */
+function gutenberg_add_post_class_to_all_post_types() {
+	$post_types = get_post_types( array( 'public' => true ), 'names' );
+
+	foreach ( $post_types as $post_type ) {
+		add_filter( "rest_prepare_{$post_type}", 'gutenberg_add_post_class_to_api_response', 10, 3 );
```

Comment:
We should use `register_rest_field` and give it an appropriate schema.

Reply:
@TimothyBJacobs I changed it to use `register_rest_field` and renamed 'post_class' to 'class_list'. Could you review the changes?